fastify.js

```diff
@@ -585,6 +585,17 @@ function build (options) {
   function addHook (name, fn) {
     throwIfAlreadyStarted('Cannot call "addHook" when fastify instance is already started!')
 
+    // TODO: v3 instead of log a warning, throw an error
+    if (name === 'onSend' || name === 'preSerialization') {
+      if (fn.constructor.name === 'AsyncFunction' && fn.length === 4) {
+        fastify.log.warn(`Async function has too many arguments. Async hooks should not use the 'next' argument.`)
+      }
+    } else {
+      if (fn.constructor.name === 'AsyncFunction' && fn.length === 3) {
+        fastify.log.warn(`Async function has too many arguments. Async hooks should not use the 'next' argument.`)
```

Comment:
This can be hard to track. I think it could be useful to include the function name. In v3 we have a stack trace.

Reply:
We can include a stack right now, this is not an async function.

Reply:
I mean when we throw an error the informations are there. We should attach the stack :+1:

Reply:
Done!